independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/EventImpl.java

```diff
@@ -339,6 +341,12 @@ private void notifyObservers(T event, ObserverExceptionHandler exceptionHandler,
                 Predicate<ObserverMethod<? super T>> predicate) {
             EventContext eventContext = new EventContextImpl<>(event, eventMetadata);
             for (ObserverMethod<? super T> observerMethod : observerMethods) {
+                Bean<?> declaringBean = observerMethod.getDeclaringBean();
+                // don't notify observers on beans with inactive context
+                if (declaringBean != null && !declaringBean.getScope().equals(Dependent.class)
+                        && Arc.container().getActiveContext(declaringBean.getScope()) == null) {
```

Comment:
I have a commit somewhere that adds a context object for the `@Dependent` pseudo-scope, which is always active. We'll be able to remove `!declaringBean.getScope().equals(Dependent.class)` when that is merged, if I understand this correctly.

Reply:
Yes, that is correct - except I will probably move this logic into the generated notify code.
I can add a comment about that.

Reply:
> I have a commit somewhere that adds a context object for the @Dependent pseudo-scope, which is always active.

Hmm, I can't say I'm a fan... why is this needed? `@Dependent` is not a context anyway.

Reply:
Actually, @Ladicek you won't need to remove anything, moving this logic into observer generator doesn't require special case handling as dependent beans always have new instances attached anyway.